virtcontainers/acrn.go

```diff
@@ -0,0 +1,630 @@
+		time.Sleep(time.Duration(50) * time.Millisecond)
+	}
+
+	// Let's try with a hammer now, a SIGKILL should get rid of the
+	// VM process.
+	return syscall.Kill(pid, syscall.SIGKILL)
+
+}
+
+func (a *acrn) udpateBlockDevice(drive *config.BlockDrive) error {
+	var err error
+	if drive.File == "" || drive.Index >= AcrnBlkDevPoolSz {
+		return fmt.Errorf("Empty filepath or invalid drive index, Dive ID:%s, Drive Index:%d",
+			drive.ID, drive.Index)
+	}
+
+	slot := AcrnBlkdDevSlot[drive.Index]
+
+	//Explicitly set PCIAddr to NULL, so that VirtPath can be used
+	drive.PCIAddr = ""
+
+	args := []string{"blkrescan", a.acrnConfig.Name, fmt.Sprintf("%d,%s", slot, drive.File)}
+
+	a.Logger().WithFields(logrus.Fields{
+		"drive": drive,
+		"path":  a.config.HypervisorCtlPath,
+	}).Infof("udpateBlockDevice with acrnctl path")
+	cmd := exec.Command(a.config.HypervisorCtlPath, args...)
+	if err := cmd.Run(); err != nil {
+		a.Logger().WithError(err).Error("updating Block device with newFile path")
+	}
+
+	return err
+}
+
+func (a *acrn) hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	span, _ := a.trace("hotplugAddDevice")
+	defer span.Finish()
+
+	switch devType {
+	case blockDev:
+		//The drive placeholder has to exist prior to Update
+		return nil, a.udpateBlockDevice(devInfo.(*config.BlockDrive))
+	default:
+		a.Logger().WithFields(logrus.Fields{"devInfo": devInfo,
+			"deviceType": devType}).Warn("hotplugAddDevice: unsupported device")
+		return nil, fmt.Errorf("hotplugAddDevice: unsupported device: devInfo:%v, deviceType%v",
+			devInfo, devType)
+	}
+}
+
+func (a *acrn) hotplugRemoveDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	span, _ := a.trace("hotplugRemoveDevice")
+	defer span.Finish()
+
+	// Not supported. return success
+
+	return nil, nil
+}
+
+func (a *acrn) pauseSandbox() error {
+	span, _ := a.trace("pauseSandbox")
+	defer span.Finish()
+
+	// Not supported. return success
+
+	return nil
+}
+
+func (a *acrn) resumeSandbox() error {
+	span, _ := a.trace("resumeSandbox")
+	defer span.Finish()
+
+	// Not supported. return success
+
+	return nil
+}
+
+// addDevice will add extra devices to Acrn command line.
+func (a *acrn) addDevice(devInfo interface{}, devType deviceType) error {
+	var err error
+	span, _ := a.trace("addDevice")
+	defer span.Finish()
+
+	switch v := devInfo.(type) {
+	case types.Volume:
+		// Not supported. return success
+		err = nil
+	case types.Socket:
+		a.acrnConfig.Devices = a.arch.appendSocket(a.acrnConfig.Devices, v)
+	case kataVSOCK:
+		// Not supported. return success
+		err = nil
+	case Endpoint:
+		a.acrnConfig.Devices = a.arch.appendNetwork(a.acrnConfig.Devices, v)
+	case config.BlockDrive:
+		a.acrnConfig.Devices = a.arch.appendBlockDevice(a.acrnConfig.Devices, v)
+	case config.VhostUserDeviceAttrs:
+		// Not supported. return success
+		err = nil
+	case config.VFIODev:
+		// Not supported. return success
+		err = nil
+	default:
+		break
```

Comment:
Shouldn't this return an `error`?

Reply:
For all the not supported cases, nil is returned. Added a log indicate failure. Both fc and qemu implementation also don't seem to be returning an error.